### PR TITLE
Fix crashes at shutdown due to references to stale debug control registry.

### PR DESCRIPTION
**Pull request by ywkaras**
Adds reference count for references to registry.  The registry is deleted only when the reference count goes to zero.

### CODE REVIEW

**Comment by ywkaras** [comment]
I think I have to make changes to make sure the registry mutex isn't used after it's destroyed.

**Comment by ywkaras** [comment]
[approve ci autest]

File: src/tscore/DbgCtl.cc [hunk 1]
```diff
@@ -84,6 +132,8 @@ DbgCtl::_get_ptr(char const *tag)
 
   auto &d{ra.data()};
 
+  ++ra.registry_reference_count;
```
Comment (wkaras) [inline src/tscore/DbgCtl.cc:135]: Increment of reference count must come before line 133.

**Review by cmcfarlen — APPROVED** [review]
This PR looks good and it does prevent crashes at shutdown.

**Pull request merged**